src/login.js

```diff
@@ -0,0 +1,54 @@
+import $ from 'jquery';
+import User from '../classes/user.js';
+import Hotel from '../classes/hotel.js';
+
+const validateLogin = (e) => {
+  e.preventDefault()
+  const username = $('.login-username').val();
+  const password = $('.login-password').val();
+
+  if (username && password === 'overlook2019') {
+    retrieveUserData(username);
+  } else {
+    console.log('Please enter value');
+  }
+}
+
+$('.login-submit').on('click', validateLogin);
+
+const linkUser = (username, userData) => {
+  const userID = username.match(/\d+/)[0];
+  const userInfo = userData.find(user => user.id === userID);
+  return new User(userInfo);
+}
+
+const loadDashboard = (username, data) => {
+  //push to a small repo 
+  const user = linkUser(username, data.users);
+  const hotel = new Hotel('Grandiose Sarajevo Motel', data.rooms, data.bookings);
+
+}
+
+const retrieveUserData = (username) => {
+  let urls = {
+    userData: 'https://fe-apps.herokuapp.com/api/v1/overlook/1904/users/users',
+    roomData: 'https://fe-apps.herokuapp.com/api/v1/overlook/1904/rooms/rooms',
+    bookingData: 'https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings',
+  }
+
+  const hotelData = {};
```

Comment:
I'm not following why you made hotelData an object, looking back at your hotel class, you have the properties of rooms and bookings set, I think that setting your data to those properties might be another good way to store your information.

Reply:
The problem with the promise.all is that it iterates, and so on each iteration the value of data within the later .then is the specific URL call it's iterating over. If I tried to instantiate a hotel class there it would instantiate it three times and each time it would only have part of the data it needs (user data, bookings, rooms). 

This was sort of my solution but it doesn't work as it strips out keys as I mentioned before. I've tried to do a couple other scrappy things but none of them work either, so I might just scrap the promise.all for the time being unless you have any quick tips!